### DISCUSSION TITLE
Failure on mvn clean install

### DISCUSSION THREAD

**Discussion by profnormano** [wgu-opensource/osmt#106] [Asking for Help] [answered]
I see these errors: 

[INFO] 04 01 2022 04:13:27.726:INFO [karma-server]: Karma v5.0.9 server started at http://0.0.0.0:9876/
[INFO] 04 01 2022 04:13:27.732:INFO [launcher]: Launching browsers ChromeHeadlessNoSandbox with concurrency unlimited
[INFO] 04 01 2022 04:13:27.734:INFO [launcher]: Starting browser ChromeHeadless
[INFO] 04 01 2022 04:13:27.735:ERROR [launcher]: No binary for ChromeHeadless browser on your platform.
[INFO]   Please, set "CHROME_BIN" env variable.

....

[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.9.1:npm (test) on project osmt-ui: Failed to run task: 'npm run ci-test' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.9.1:npm (test) on project osmt-ui: Failed to run task

Did I miss something in setup?  I've installed Maven 3.8.4 on an AWS EC2/t3.large, node v17.3.0; openjdk v1.8.0_312

Any suggestions would be appreciated.

**Comment by profnormano** [comment]
I found this advice and I'll test it later this morning:  

I had to run this on build server:

wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -
echo "deb http://dl.google.com/linux/chrome/deb/ stable main" > /etc/apt/sources.list.d/google.list
apt-get update
apt-get install -y google-chrome-stable xvfb
npm i -g @angular/cli

credits: https://stackoverflow.com/questions/65837359/docker-container-gives-no-binary-for-chrome-browser-on-your-platform

**Comment by profnormano** [comment, marked as answer]
This was the better set of commands for AWS EC2 setup:

wget https://dl.google.com/linux/direct/google-chrome-stable_current_x86_64.rpm
sudo yum install ./google-chrome-stable_current_x86_64.rpm
sudo ln -s /usr/bin/google-chrome-stable /usr/bin/chromium

credits: https://help.looker.com/hc/en-us/articles/360035411973-Installing-Chromium-for-Amazon-Linux#:~:text=Install%20Chrome%20by%20running%3A%20sudo%20yum%20install.%2Fgoogle-chrome-stable_current_x86_64.rpm%20Looker,create%20a%20symbolic%20link%20from%20there%20to%20Chrome

**Comment by JohnKallies** [comment]
That's good insight.

Should we be running a client build on an EC2? What do you think about creating a package/artifact of the static UI files, and deploying those without putting build tooling on the instance? As-is, the OSMT build puts the static files in the resources directory in the api module, making a pretty flat deployment. I'm not a fan of that, and I'd rather see the UI build be more S3 friendly.

FYI, we can skip test in Maven with `mvn clean install -DskipTests`, and skip tests that rely on Docker/Chrome with `mvn clean install -P dockerfile-build`

I can't completely tell from the context... are you asking for something? It looks like you figured it out and marked the answer.

**Reply by profnormano** [reply]
Yes, I answered my own question above.    I'm using EC2 as my development instance to work through the install instructions.  Later I can look for optimization and yes, S3 for static files would be more friendly and make it easier to push them out through AWS CloudFront to get a global CDN.
